src/capture-view.ts

```diff
@@ -77,8 +77,12 @@ export default class CaptureView extends View {
   }
 
   async getDevices() {
-    let devices = await navigator.mediaDevices.enumerateDevices() as MediaDeviceInfo[];
-    devices = devices.filter((device) => device.kind === 'videoinput');
+    let devices: MediaDeviceInfo[] = [];
+
+    if ('mediaDevices' in navigator) {
```

Comment:
Done - a little lonely right now, but there is more I can pull out like this as I go along.